programs/disks/disks.cpp

```diff
@@ -0,0 +1,354 @@
+#include <stdlib.h>
+#include <fcntl.h>
+#include <map>
+#include <iostream>
+#include <iomanip>
+#include <optional>
+#include <filesystem>
+#include <string>
+
+#include <boost/program_options.hpp>
+
+#include <base/argsToConfig.h>
+
+#include <Common/TerminalSize.h>
+
+#include <Columns/ColumnString.h>
+
+#include <Formats/registerFormats.h>
+
+#include <Poco/Util/HelpFormatter.h>
+
+#include <Disks/registerDisks.h>
+#include <Disks/IDisk.h>
+
+#include "disks.h"
+
+namespace fs = std::filesystem;
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int BAD_ARGUMENTS;
+}
+
+std::string DisksFunc::getDefaultConfigFileName() const
+{
+    return "/etc/clickhouse-server/config.xml";
+}
+
+void DisksFunc::listDisks()
+{
+    while(argumentsToDo[0] != "list-disks")
+        argumentsToDo.erase(argumentsToDo.begin());
+
+    argumentsToDo.erase(argumentsToDo.begin());
+
+    if (int(argumentsToDo.size()) == 0)
+    {
+        std::vector<std::string> disksNames;
+
+        for (auto & [disk_name, disk] : global_context->getDisksMap())
+        {
+            disksNames.push_back(disk->getName());
+        }
+
+        std::cout << "Disks names: \n";
+        for (auto it = disksNames.begin(); it != disksNames.end(); it++)
+        {
+            std::cout << *it << "\n";
+        }
+    }
+    else
+    if (int(argumentsToDo.size()) > 0)
+    {
+        curCommand = "list-disks";
+        init(argumentsToDo);
+        run();
+    }
+}
+
+void DisksFunc::listFiles()
+{
+    if (curCommand == "start")
+    {
+        if (int(argumentsToDo.size()) == 1)
+        {
+            printHelpMessage();
+            throw DB::Exception("Bad Arguments", DB::ErrorCodes::BAD_ARGUMENTS);
+        }
+
+        while (argumentsToDo[0] != "ls")
+            argumentsToDo.erase(argumentsToDo.begin());
+
+        argumentsToDo.erase(argumentsToDo.begin());
+
+        curCommand = "ls";
+        init(argumentsToDo);
+        run();
+    }
+    else
+        if (curCommand == "ls")
+        {
+            if (int(argumentsToDo.size()) > 1)
+            {
+                printHelpMessage();
+                throw DB::Exception("Bad Arguments", DB::ErrorCodes::BAD_ARGUMENTS);
+            }
+
+            diskPath.parsPath(in[0]);
+
+            std::vector<std::string> file_names;
+
+            DiskPtr disk_test = global_context->getDisk(diskPath.getDiskName());
+
+            std::string full_path = disk_test->getPath();
+
+            if (diskPath.getDiskName() == "default")
+                full_path = "/var/lib/clickhouse/data/default/";
+
+            full_path.pop_back();
+            full_path += diskPath.getPath();
+
+            disk_test->listFiles(full_path, file_names);
+
+            for (auto iter = file_names.begin(); iter != file_names.end(); iter++)
+            {
+                std::cout << *iter << "\n";
+            }
+        }
+}
+
+void DisksFunc::reloadConfiguration()
```

Comment:
This method is called only once on initializtion, so it's better loadCondfiguration, but I think you can just inline it in caller